packages/components/src/components/radio-group/radio-group.jsx

```diff
@@ -1,26 +1,53 @@
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import React from 'react';
-import Radio from './radio.jsx';
 
-class RadioGroup extends React.PureComponent {
+class RadioGroup extends React.Component {
+    state = {
+        selectedOption: this.props.defaultOption,
+    };
+
+    // componentDidUpdate(prevProps) {
+    //     if (this.props.defaultChecked === prevProps.defaultChecked) {
+    //         return;
+    //     }
+
+    //     this.setState({ checked: this.props.defaultChecked });
+    // }
+
+    onChange(e) {
+        this.setState({
+            selectedOption: e.target.value,
+        });
+        this.props.onToggle(e);
+    }
+
     render() {
-        const { selected, items, className } = this.props;
+        const { items, className, name } = this.props;
+
         return (
             <div className={classNames('dc-radio-group', className)}>
                 {items.map((item, idx) => (
-                    <Radio
-                        key={idx}
-                        id={item.id}
-                        value={item.value}
-                        defaultChecked={item.value === selected}
-                        onChange={this.props.onToggle}
-                        className={item.className}
-                        name={this.props.name}
-                        disabled={item.disabled}
-                    >
-                        {item.label}
-                    </Radio>
+                    <label key={idx} className={classNames('dc-radio-group__item', className)}>
+                        <input
```

Comment:
Why radio component is being removed here?

Reply:
They are the same component `Radio` and `Radio Goup` and we never use <Radio> separately, but it caused problem for the state of the selected radio when we had variables instead of a boolean variable. So, after talking to Mahdi and trying a solution he let me to remove this extra component.